shared/js/web_manifest_helper.js

```diff
@@ -0,0 +1,217 @@
+/* exported WebManifestHelper */
+'use strict';
+
+/**
+ *  Helper to fetch and process a Web Manifest.
+ */
+(function WebManifestHelper(exports) {
+
+  /**
+   * Get Manifest.
+   *
+   * @param {String} url Manifest URL
+   * @returns {Promise} A promise of a response containing manifest data.
+   */
+  var getManifest = function(url) {
+    return new Promise(function(resolve, reject) {
+      var xhr = new XMLHttpRequest({mozSystem: true});
```

Comment:
I want to use the ```LazyLoader.getJSON``` method here instead. The only thing I don't think it supports yet is ```setRequestHeader```. We might need to extend it to do so if this is required.